packages/react-dom/src/__tests__/ReactDOMComponent-test.js

```diff
@@ -2506,6 +2506,32 @@ describe('ReactDOMComponent', () => {
     });
   });
 
+  describe('Boolean attributes', function() {
+    it('warns on the ambiguous string value "false"', function() {
+      let el;
+      expect(() => {
+        el = ReactTestUtils.renderIntoDocument(<div hidden="false" />);
+      }).toWarnDev(
+        'Received the string `false` for the boolean attribute `hidden`. ' +
+          'Did you mean hidden={false}?',
+      );
+
+      expect(el.getAttribute('hidden')).toBe('');
+    });
+
+    it('warns on the ambiguous string value "False"', function() {
+      let el;
+      expect(() => {
+        el = ReactTestUtils.renderIntoDocument(<div hidden="False" />);
```

Comment:
Since title case `False` doesn’t have special meaning I don’t think it’s worth testing for.

Reply:
Happy to take your steer on this, I'd just like to state my case first for completeness: The thing I was trying to make safer was people drawing inferences from one "boolean" HTML attribute to another. Since [enumerated attribute values are case-insensitive](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute), `spellCheck="False"` does behave differently to `hidden="False"`, in exactly the same way as the respective lowercase variants do. I take your point that most people would be using lowercase to begin with, so this is probably a rarer case, but it does theoretically occur.

To clarify, you're referring to dropping the `toLowerCase()` in the actual check, as well as removing this test, right?

Reply:
Yeah. I see your point but I think it's unlikely enough to be written accidentally and we want to make the checks as cheap as possible. Strict comparison is better than extra `toLowerCase` call.